backend-rust/migrations/0001_initialize.up.sql

```diff
@@ -582,3 +594,45 @@ $trigger$ LANGUAGE plpgsql;
 CREATE TRIGGER account_updated_notify_trigger AFTER INSERT
 ON affected_accounts
 FOR EACH ROW EXECUTE PROCEDURE account_updated_notify_trigger_function();
+
+-- Table for logging all account-related activities on-chain.
+-- This table tracks individual entries related to changes in account balances.
+CREATE TABLE account_statements (
+    -- Unique identifier for each account statement entry.
+    id
+        BIGINT
+        GENERATED ALWAYS AS IDENTITY
+        PRIMARY KEY,
+    -- Index of the account associated with this entry.
+    account_index
+        BIGINT
+        REFERENCES accounts(index)
+        NOT NULL,
+    -- Type of the account statement entry.
+    entry_type
+        account_statement_entry_type
+        NOT NULL,
+    -- Amount associated with the entry in micro CCD.
+    -- This represents the change in balance caused by the transaction or activity.
+    -- Will be negative when an amount is being subtracted from the account.
+    amount
+        BIGINT
+        NOT NULL,
+    -- The resulting balance of the account after applying this entry.
+    -- This is used to track the account's total CCD at the time of the transaction.
+    account_balance
+        BIGINT
+        NOT NULL,
+    -- Block height at which the entry occurred.
+    -- Links to the blocks table to associate the entry with a specific block.
+    block_height
+        BIGINT
+        REFERENCES blocks(height)
+        NOT NULL,
+    -- Used as reference for all account statements not of type reward type
+    transaction_id
+        BIGINT
+        NULL
+);
+
+CREATE INDEX account_statements_entry_type_idx ON account_statements (account_index, entry_type);
```

Comment:
Don't we want the `id` as part of the index as well?
```suggestion
CREATE INDEX account_statements_entry_type_idx ON account_statements (account_index, entry_type, id);
```

Reply:
```

                    AND id > $1
                    AND id < $2
```

We only need because of that, and such index already exists because of the field being primary key

Reply:
I won't be able to use the primary key index because of the other WHERE statements in the query